utils/firestore.js

```diff
@@ -2987,23 +2987,50 @@ const processEventWebhook = async (event) => {
         }
     } else {
         const eventRecord = {
-            [`${event.event}_status`]: true,
-            [`${event.event}_date`]: date,
-            [`${event.event}_timestamp`]: event.timestamp,
-            connect_id: event.connect_id,
+            [`${event.event}Status`]: true,
+            [`${event.event}Date`]: date,
+            [`${event.event}Timestamp`]: event.timestamp,
+            connectId: event.connect_id,
             email: event.email,
-            notification_id: event.notification_id,
-            sg_event_id: event.sg_event_id,
-            sg_message_id: event.sg_message_id,
+            notificationId: event.notification_id,
+            sgEventId: event.sg_event_id,
+            sgMessageId: event.sg_message_id,
             token: event.token,
         };
         if (["bounce", "dropped"].includes(event.event)) {
-            eventRecord[`${event.event}_reason`] = event.reason;
+            eventRecord[`${event.event}Reason`] = event.reason;
         }
         await db.collection("sendgridTracking").add(eventRecord);
     }
 };
 
+const processTwilioEvent = async (event) => {
+
+    const date = new Date().toISOString();
+    console.log(`SID: ${event.MessageSid}, Status: ${event.MessageStatus}`);
+
+    const snapshot = await db
+        .collection("notifications")
+        .where("messageSid", "==", event.MessageSid)
+        .get();
+
+    if (snapshot.size > 0) {
+        for (let doc of snapshot.docs) {
```

Comment:
Each `messageSid` should be unique across all notification records. Given a `messageSid`, there should be 0 or 1 document matches.  Thus this for loop may not be necessary.